frontend/src/components/fileupload.js

```diff
@@ -63,14 +71,14 @@ export class FileUpload extends React.Component {
 
   render() {
     const { children, className } = this.props;
-    const Component = this.props.component || 'button';
     const styles = this.props.isUnstyled ? {} : {cursor: 'pointer', display: 'inline', padding: '0', margin: '0'};
+    const { primaryObject } = this.context;
     return (
       <React.Fragment>
         <input type="file" multiple={this.state.multiple} style={{display: 'none'}} onChange={this.onFileChange} ref={this.inputRef} />
-        <Component className={className} style={styles} onClick={this.onUploadClick}>
+        <Button className={className} style={styles} onClick={this.onUploadClick} isDisabled={!primaryObject}>
```

Comment:
I changed this to use `isAriaDisabled` so that it would remain focus-able for accessibility.

The button has been unstyled in ibutsu, and because it's in the header with a plain variant specified, I would agree it's not easy to see that it is disabled until you mouse over it.

Going to commit what I have with the tooltip to provide context, let me know if the styling when disabled is a blocking issue for you and we can discuss it further.